### DISCUSSION TITLE
Migrating to Quarkus 18.0.0 version

### DISCUSSION THREAD

**Discussion by sreekesh93** [keycloak/keycloak#14524] [Archive]
We have 18..0.0 legacy keycloak currently if i am moving it to Quarkus version, is there any data migration happens?
 We deployed   quay.io/keycloak/keycloak:18.0.0-legacy on top of this we are migrating towards quay.io/keycloak/keycloak:18.0.0 which is the quarkus version of keycloak.
We could see all data created in legacy in quarkus, and to check backward compatibility we re-installed legacy version of keycloak which also works fine.

Question here is, Is data migration take place when we migrate keycloak from legacy to quarkus version?

Database used: Postgres

@stianst  @andymunro

**Comment by sschu** [comment]
AFAIK there are no special migrations only for the Quarkus version.

**Comment by sreekesh93** [comment]
Then I think there is a issue with this documentation, https://www.keycloak.org/docs/latest/upgrading/ here it stated that ```The database will no longer be compatible with the old server after the upgrade```

**Reply by sschu** [reply]
This refers to a Keycloak version upgrade. Switching to Keycloak.X with the same version is not a version upgrade.

**Reply by sreekesh93** [reply]
Yes we are migrating from legacy version (wildfly) to quarkus version of keycloak. So is the data migration expected in this case?

**Reply by sschu** [reply]
Not if you don't change versions, just switch between Wildfly and Keycloak.X with the same Keycloak version.
